pkg/model/model.go

```diff
@@ -633,6 +410,13 @@ func (p *ProcessPipeline) Process(process Processor) Pipeline {
 	return NewProcessPipeline(p, process)
 }
 
+func (p *ProcessPipeline) WithSource(source Source) Pipeline {
+	if s, ok := p.source.(*ProcessPipeline); ok {
+		return &ProcessPipeline{p.collector, s.WithSource(source).(Source), p.Processor, nil}
```

Comment:
is it safe to share `collector` object ?

Reply:
No :) i will create a new Collector, and check if i can improve the ProcessPipeline duplication